addon/services/plaid.js

```diff
@@ -0,0 +1,52 @@
+import { Promise } from 'rsvp';
+import Service from '@ember/service';
+
+export default Service.extend({
+  isLoading: false,
+  isInjected: false,
+  plaidPromise: null,
+
+  /**
+   * This method loads the Plaid the script. It can be safely called multiple times. If the script
+   *   fails to load and the method is called again, the method will reattempt to load script.
+   * @returns {Promise} a promise which resolves when the Plaid script loads and rejects when it
+   *   does not.
+   * @public
+   */
+  injectScript() {
+    if (this.get('isLoading') || this.get('isInjected')) {
```

Comment:
if `isInjected` is true, can't we return a no-op instead?

Reply:
The method returns a promise. It would be inconsistent to return a promise in some places and not others.